graylog2-web-interface/src/views/logic/views/Actions.js

```diff
@@ -0,0 +1,15 @@
+// @flow strict
+import history from 'util/History';
+import Routes from 'routing/Routes';
+
+export const loadNewView = () => {
+  return history.push(`${Routes.SEARCH}/new`);
+};
+
+export const loadNewViewForStream = (streamId: string) => {
+  return history.push(`${Routes.stream_search(streamId)}/new`);
+};
+
+export const loadView = (viewId: string) => {
+  return history.push(`${Routes.SEARCH}/${viewId}`);
```

Comment:
There is still a `history.push` in `SavedSearchControls:162` which does the same and is no longer needed. I found it because I got the warning `Saving view failed: TypeError: Cannot read property 'then' of undefined` after creating a new saved search.

`loadView` returned a Promise before these changes, but I could not find another case where this could be a problem.

Reply:
:white_check_mark: